### ISSUE TITLE
Server Runtime forced on JDK8 which is not compatible with Jetty and JSP pages

### ISSUE THREAD

**Issue by NicolaSpreafico** [GoogleCloudPlatform/google-cloud-eclipse#590]
Hello,
as already established from a [SO thread](http://stackoverflow.com/questions/38030398/google-app-engine-java-8-and-jsp-files-compilation-error), the current Jetty version does not handle the JSP pages using JDK8 as runtime, so it is required to force JDK7 inside the Server Runtime.

I have JDK8 installed because is strictly required as runtime for my Eclipse Neon version.

With the previous Google Plugin this goal was easily gettable, just forcing a specific JDK version during the creation of the Server Runtime
![image](https://cloud.githubusercontent.com/assets/12278553/17811811/a9c0c774-6624-11e6-84cf-c1880f3cc179.png)

With the current `gcloud-eclipse-tools` it is no more possible select the specific JDK version to use with the runtime (and as the same way the App Engine SDK, but I don't care about that).

It is clear when Jetty (GAE dev-server) starts on JDK8 because as soon I try to access a JSP page I get the following error and stacktrace
```
HTTP ERROR 500

Problem accessing /. Reason:

    Unable to compile class for JSP: 

An error occurred at line: 1 in the generated java file
The type java.io.ObjectInputStream cannot be resolved. It is indirectly referenced from required .class files

Stacktrace:
Caused by:

org.apache.jasper.JasperException: Unable to compile class for JSP: 

An error occurred at line: 1 in the generated java file
The type java.io.ObjectInputStream cannot be resolved. It is indirectly referenced from required .class files

Stacktrace:
	at org.apache.jasper.compiler.DefaultErrorHandler.javacError(DefaultErrorHandler.java:92)
	at org.apache.jasper.compiler.ErrorDispatcher.javacError(ErrorDispatcher.java:330)
	at org.apache.jasper.compiler.JDTCompiler.generateClass(JDTCompiler.java:439)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:349)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:327)
	at org.apache.jasper.compiler.Compiler.compile(Compiler.java:314)
	at org.apache.jasper.JspCompilationContext.compile(JspCompilationContext.java:592)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:317)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:313)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:260)
	at com.google.appengine.tools.development.PrivilegedJspServlet.access$101(PrivilegedJspServlet.java:23)
	at com.google.appengine.tools.development.PrivilegedJspServlet$2.run(PrivilegedJspServlet.java:61)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.google.appengine.tools.development.PrivilegedJspServlet.service(PrivilegedJspServlet.java:58)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:717)
```

This problem (with the previous plugin) was already solved forcing, as already mentioned, JDK7 as Server Runtime.

With the newer plugin, when I create the runtime no other options are available. I can only create the runtime as itself.
![image](https://cloud.githubusercontent.com/assets/12278553/17812200/4e8a1e62-6626-11e6-941f-eef0cccacffa.png)

Also, the EDIT button is grayed out
![image](https://cloud.githubusercontent.com/assets/12278553/17812216/5d54a570-6626-11e6-970b-d2eb2d5854c2.png)

How can i declare the specific JDK runtime to use with the local dev-server?

Thanks
Regards

**Comment by briandealwis** [comment]
Thanks for the feedback.  I think you might be able to work around this by setting `JAVA_HOME` to point to your Java 7 JRE, before launching Eclipse.  We're not setting `JAVA_HOME` so it should be passed through.

**Comment by briandealwis** [comment]
I dug into this a bit.  The problem here is that the Google Cloud SDK's `dev_appserver` is based on an old version of Jetty that uses an old version of the Eclipse Java Compiler (ECJ) for compiling the JSPs.  This old ECJ pre-dates Java 8 and doesn't know how to read the class files from the Java 8 JRE libraries.

Fortunately later versions of ECJ are binary compatible and you can fix this with a bit of delicate surgery within the Google Cloud SDK using steps I outlined on that [StackOverflow question](http://stackoverflow.com/a/39050364/600339).

**Comment by NicolaSpreafico** [comment]
Hello @briandealwis 

At first I tried to configure the `JAVA_HOME` variable.
![image](https://cloud.githubusercontent.com/assets/12278553/17845569/2aba088e-6843-11e6-9f35-ab556ef47338.png)
![image](https://cloud.githubusercontent.com/assets/12278553/17845581/3a11c10a-6843-11e6-86a0-a2ae49f39897.png)

Before of that I did not have any already existing `JAVA_HOME` variable, so I don't think this configuration will create problems on my PC. The server started correctly (I have other library problems but I will open another issue for that)

On the other hand... I also tried the other approach (edit the JAR files inside the CloudSDK/AppEngineSDK folder) and it works as the same way.

Until there will be a more "high level" approach, I will use the JAVA_HOME setting, because it is more simplier to configure. I fear that at every SDK update I will need to update again the JAR files, so it is not very comfortable to do this.

**Comment by elharo** [comment]
b/29462371

expecting a fix from the dependency by the end of the month

**Comment by elharo** [comment]
Still waiting on the dependency to be fixed, but it is in progress.

**Comment by elharo** [comment]
Expecting gcloud 131.0.0. to fix this on release.

**Comment by elharo** [comment]
Fixed as of Cloud SDK 131.0
